### PR TITLE
Added a script to help create a default ceph storageclass

### PR DESCRIPTION
**Pull request by gyliu513**
@stacyped ^^

### CODE REVIEW

**Review by taylormgeorge91 — COMMENTED** [review]
May want to allow an option/arg for the default annotation.

If a default storage class already exists, you have to remove the annotation from that one.
This may be something that should be controlled through an arg instead of done all the time.

**Comment by taylormgeorge91** [comment]
The yum upgrade will only work on certain OS.

File: scripts/setup-ceph-storage.sh [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  fi
+}
+
+function create_rook_storagepool() {
+  printf "\n====> Create rook StoragePool\n"
+
+  $( oc create -f rook/cluster/examples/kubernetes/ceph/csi/rbd/storageclass-test.yaml  >/dev/null 2>&1 )
+
+  if [ $? -eq 0 ]; then
+    printf "<==== Rook stroagepool setup successfully \n"
```
Comment (taylormgeorge91) [inline scripts/setup-ceph-storage.sh:197]: storagepool

File: scripts/setup-ceph-storage.sh [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  clone_rook
+  setup_rook
+
+  printf "====> Waiting 5 minutes for Ceph/Rook to spin up\n\n"
+  sleep 300
+
+  check_ceph_state
+}
+
+printf "This script will help you set up a default storageclass via Rook/Ceph\n\n"
```
Comment (taylormgeorge91) [inline scripts/setup-ceph-storage.sh:252]: Note it uses the fs type one which supports RWX

File: scripts/setup-ceph-storage.sh [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+read -p "Input the target disk that you want to use for Ceph: " disk
+main $disk
```
Comment (taylormgeorge91) [inline scripts/setup-ceph-storage.sh:254]: How does one determine the disk value to use?

Will there be associated guidance to determine that value?

**Comment by taylormgeorge91** [comment]
Noticed these when running:

```
====> Updating OS. This might take a spell
<==== Yum update did not succeed

./rook-ceph-setup.sh: line 76: determine_system_arch: command not found
====> Checking if git is installed
<==== git is installed
./rook-ceph-setup.sh: line 91: login_oc_cmdline: command not found
```

File: scripts/setup-ceph-storage.sh [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+main(){
+
+  initialize $1
+  preamble
+  clone_rook
+  setup_rook
+
+  printf "====> Waiting 5 minutes for Ceph/Rook to spin up\n\n"
+  sleep 300
```
Comment (taylormgeorge91) [inline scripts/setup-ceph-storage.sh:247]: Recommend polling instead of sleep

**Comment by stacyped** [comment]
Please let me know once changes have been made and it is oK to merge

**Comment by gyliu513** [comment]
thanks @stacyped and @taylormgeorge91 , let me review and update later

**Comment by gyliu513** [comment]
we will not need this one, but will use some yaml templates to deploy a default storageclass for PoC.

**Pull request closed without merging**